CMakeLists.txt

```diff
@@ -6,28 +6,58 @@ endif()
 
 cmake_minimum_required(VERSION 3.0)
 project(Luau LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 17)
 
 option(LUAU_BUILD_CLI "Build CLI" ON)
+option(LUAU_BUILD_WEB_REPL "Build Web CLI" ON)
 option(LUAU_BUILD_TESTS "Build tests" ON)
 
 add_library(Luau.Ast STATIC)
 add_library(Luau.Compiler STATIC)
 add_library(Luau.Analysis STATIC)
 add_library(Luau.VM STATIC)
 
-if(LUAU_BUILD_CLI)
-    add_executable(Luau.Repl.CLI)
-    add_executable(Luau.Analyze.CLI)
+if(LUAU_BUILD_WEB_REPL AND EMSCRIPTEN)
+    # this is a hack, disable cli and tests since we're compiling using emscripten
+    set(LUAU_BUILD_CLI OFF)
+    set(LUAU_BUILD_TESTS OFF)
 
-    # This also adds target `name` on Linux/macOS and `name.exe` on Windows
-    set_target_properties(Luau.Repl.CLI PROPERTIES OUTPUT_NAME luau)
-    set_target_properties(Luau.Analyze.CLI PROPERTIES OUTPUT_NAME luau-analyze)
-endif()
+    add_definitions(-DLUAU_WEB_REPL)
```

Comment:
We shouldn't be setting global options, instead preferring the per-target specification like we do for other targets